src/inc/handlers.ts

```diff
@@ -130,7 +130,10 @@ export const beforeContentReplace: Handler<'content:replace'> = function (
 	const cache = this.swup.cache.get(visit.to.url);
 	if (!cache || !cache.fragmentHtml) return;
 
+	visit.to.document = new DOMParser().parseFromString(cache.fragmentHtml, 'text/html');
+	visit.to.html = cache.fragmentHtml;
 	args.page.html = cache.fragmentHtml;
```

Comment:
@daun this is currently only here for backwards compatibility. We don't want that, as it would inflate the bundle size over time, right? What's the best way to make sure then that fragment plugin will always have swup 4.6.0 available? Set the peerDependency to that version?